### PR TITLE
Add DPS client SDK to list of available SDKs

### PR DESCRIPTION
**Pull request by BryanLa**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 This repository contains the following:
 * **Microsoft Azure IoT Hub device SDK for C** to connect devices running C code to Azure IoT Hub
+* **Microsoft Azure Iot Hub Device Provisioning Service client SDK** for enrolling devices with Device Provisioning Services
```
Comment (yzhong94) [inline readme.md]: Can we add: for enrolling devices with Device Provisioning Services and managing enrollment lists.

**Comment by BryanLa** [comment]
@yzhong94  - done!

**Comment by yzhong94** [comment]
Thanks @BryanLa !  Can you stash the commits so there's only 1 commit?  This makes the release notes clearer.  I will pull this in once it's done.

**Comment by BryanLa** [comment]
@yzhong94 - do you mean squash? If so, do you know how I can do this for a GitHub-generated patch branch?

**Comment by yzhong94** [comment]
Yeah I mean squash sorry!  I think you'd need to clone your patch branch to your desktop and do it from there.

**Comment by BryanLa** [comment]
Hi @yzhong94 - done. I had to move my work to a different branch, so I've replaced this PR with PR #368: https://github.com/Azure/azure-iot-sdk-c/pull/368. Apologies to all if this messed up the review workflow :/

Btw, have you considered turning this feature on in the repo settings, so the PR review/merge team can do: https://help.github.com/articles/configuring-commit-squashing-for-pull-requests/?

**Comment by yzhong94** [comment]
Hmm this is a really useful feature!  I have squashing too... @jspaith @tameraw  can we do this?

**Comment by jspaith** [comment]
Yes to squashing at gate.  This is on our radar and totally agree there's value there.

**Pull request closed without merging**